File: app/src/main/java/com/example/stas/oxforddictionary/presentation/view/main/MainActivity.java
package com.example.stas.oxforddictionary.presentation.view.main;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.design.widget.NavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

import com.example.stas.oxforddictionary.App;
import com.example.stas.oxforddictionary.R;
import com.example.stas.oxforddictionary.presentation.view.base.BaseActivity;
import com.example.stas.oxforddictionary.presentation.view.entry.EntryFragment;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends BaseActivity implements MainContract.View, IMainActivity {
    @BindView(R.id.nav_view)
    NavigationView navigationView;
    @BindView(R.id.drawer_layout)
    DrawerLayout drawerLayout;
    @BindView(R.id.hamburgerBtn)
    ImageButton hamburgerBtn;
    private TextView allWordsCount;
    @Inject
    MainContract.Presenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        ((App)getApplication()).createMaincomponent().injectMainActivity(this);
      // ApplicationComponentKt.getMyApplication(this).createMaincomponent().injectMainActivity(this);
     /*   App.getInstance().getMainComponent().injectMainActivity(this);*/
        presenter.attachView(this);
        init();

        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                menuItem.setChecked(true);
                drawerLayout.closeDrawers();
                //implement swaping fragments here
                return true;
            }
        });
    }

    @Override
    public void inflateFragment(String fragmentTag, String word) {
        /*if (fragmentTag.equals(getString(R.string.fragment_synonym))){
            SynonymFragment synonymFragment = new SynonymFragment();
            doFragmentTransaction(synonymFragment, fragmentTag, true, word);
        }*/
    }

    private void init() {
        EntryFragment entryFragment = new EntryFragment();
        doFragmentTransaction(entryFragment, getString(R.string.fragment_entry), false, "");
        allWordsCount = navigationView.getHeaderView(0).findViewById(R.id.allWordsCount);
    }

    private void doFragmentTransaction(Fragment fragment, String tag, boolean addToBackStack, String message) {
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        if (!message.equals("")) {
            Bundle bundle = new Bundle();
            bundle.putString(getString(R.string.intent_message), message);
            fragment.setArguments(bundle);
        }
        transaction.replace(R.id.content_frame, fragment, tag);
        if (addToBackStack) {
            transaction.addToBackStack(tag);
        }
        transaction.commit();
    }

    @Override
    public void navigateToSynonyms(Context context, String wordId) {
        this.navigator.navigateToSynonyms(context, wordId);
    }

    @Override
    public void navigateToExamples(Context context, String wordId) {
        this.navigator.navigateToExamples(context, wordId);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        presenter.dettachView();
    }

    public void openDrawer(View view) {
        drawerLayout.openDrawer(GravityCompat.START);
    }
}

File: app/src/main/java/com/example/stas/oxforddictionary/presentation/view/main/MainActivity.kt
package com.example.stas.oxforddictionary.presentation.view.main

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.view.GravityCompat
import android.view.View
import android.widget.TextView
import com.example.stas.oxforddictionary.App
import com.example.stas.oxforddictionary.R
import com.example.stas.oxforddictionary.presentation.view.base.BaseActivity
import com.example.stas.oxforddictionary.presentation.view.entry.EntryFragment
import com.example.stas.oxforddictionary.presentation.view.practice.PracticeFragment
import com.example.stas.oxforddictionary.presentation.view.save.SaveFragment
import com.example.stas.oxforddictionary.presentation.view.statistic.StatisticFragment
import kotlinx.android.synthetic.main.activity_main.*
import java.lang.Exception
import javax.inject.Inject

class MainActivity : BaseActivity(), MainContract.View, IMainActivity {

    private var allWordsCount: TextView? = null
    @Inject
    lateinit var presenter: MainContract.Presenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        (application as App).createMaincomponent().injectMainActivity(this)
        presenter.attachView(this)
        init()

        navView.setNavigationItemSelectedListener { menuItem ->
            menuItem.isChecked = true
            drawer.closeDrawers()
            inflateFragment(menuItem.title.toString())
            true
        }
    }

    private fun inflateFragment(fragmentTag: String, message: String = "")  {
        when(fragmentTag){
            getString(R.string.menu_search).toString() -> doFragmentTransaction(EntryFragment(), fragmentTag, false, message)
            getString(R.string.menu_search_history).toString() -> Unit
            getString(R.string.menu_training).toString() ->  doFragmentTransaction(PracticeFragment(), fragmentTag, false, message)
            getString(R.string.menu_saved).toString() ->   doFragmentTransaction(SaveFragment(), fragmentTag, false, message)
            getString(R.string.menu_settings).toString() -> doFragmentTransaction(StatisticFragment(),fragmentTag, false, message)
            getString(R.string.menu_author_about).toString() -> sendToAuthor()
            else -> throw RuntimeException("Something went wrong")
        }
    }

    private fun sendToAuthor(){
        val emailIntent = Intent(Intent.ACTION_SEND)
        emailIntent.type = "text/plain"
        emailIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf("StasStahmaJob@gmail.com"))
        emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Отклик Пользователя")
        startActivity(Intent.createChooser(emailIntent, "Send Email..."))
    }



    private fun init() {
        val entryFragment = EntryFragment()
        doFragmentTransaction(entryFragment, getString(R.string.fragment_entry), false, "")
        allWordsCount = navView.getHeaderView(0).findViewById(R.id.allWordsCount)
    }



    private fun doFragmentTransaction(fragment: Fragment, tag: String, addToBackStack: Boolean, message: String) {
        val transaction = supportFragmentManager.beginTransaction()
        if (message != "") {
            val bundle = Bundle()
            bundle.putString(getString(R.string.intent_message), message)
            fragment.arguments = bundle
        }
        transaction.replace(R.id.content_frame, fragment, tag)
        if (addToBackStack) {
            transaction.addToBackStack(tag)
        }
        transaction.commit()
    }

    override fun navigateToSynonyms(context: Context?, wordId: String) {
        navigator.navigateToSynonyms(context, wordId)
    }

    override fun navigateToExamples(context: Context?, wordId: String) {
        navigator.navigateToExamples(context, wordId)
    }

    override fun navigateToPractice(context: Context?, wordId: String) {
        navigator.navigateToPractice(context, wordId)
    }

    override fun onDestroy() {
        super.onDestroy()
        presenter.dettachView()
    }

    fun openDrawer(view: View) {
        drawer.openDrawer(GravityCompat.START)
    }
}
